install.sh

```diff
@@ -36,46 +45,95 @@ if [ $dbtype == "postgresql" ];then
+
+hdfsRootUser=$(charEscape ${hdfsRootUser})
 sed -i ${txt} "s@^hdfs.root.user=.*@hdfs.root.user=${hdfsRootUser}@g" conf/common.properties
+
+defaultFS=$(charEscape ${defaultFS})
 sed -i ${txt} "s@^fs.defaultFS=.*@fs.defaultFS=${defaultFS}@g" conf/common.properties
+
+s3Endpoint=$(charEscape ${s3Endpoint})
 sed -i ${txt} "s@^fs.s3a.endpoint=.*@fs.s3a.endpoint=${s3Endpoint}@g" conf/common.properties
+
+s3AccessKey=$(charEscape ${s3AccessKey})
 sed -i ${txt} "s@^fs.s3a.access.key=.*@fs.s3a.access.key=${s3AccessKey}@g" conf/common.properties
+
+s3SecretKey=$(charEscape ${s3SecretKey})
 sed -i ${txt} "s@^fs.s3a.secret.key=.*@fs.s3a.secret.key=${s3SecretKey}@g" conf/common.properties
+
+resourceManagerHttpAddressPort=$(charEscape ${resourceManagerHttpAddressPort})
 sed -i ${txt} "s@^resource.manager.httpaddress.port=.*@resource.manager.httpaddress.port=${resourceManagerHttpAddressPort}@g" conf/common.properties
+
+yarnHaIps=$(charEscape ${yarnHaIps})
 sed -i ${txt} "s@^yarn.resourcemanager.ha.rm.ids=.*@yarn.resourcemanager.ha.rm.ids=${yarnHaIps}@g" conf/common.properties
+
+singleYarnIp=$(charEscape ${singleYarnIp})
 sed -i ${txt} "s@^yarn.application.status.address=.*@yarn.application.status.address=http://${singleYarnIp}:%s/ws/v1/cluster/apps/%s@g" conf/common.properties
 sed -i ${txt} "s@^yarn.job.history.status.address=.*@yarn.job.history.status.address=http://${singleYarnIp}:19888/ws/v1/history/mapreduce/jobs/%s@g" conf/common.properties
+
+sudoEnable=$(charEscape ${sudoEnable})
 sed -i ${txt} "s@^sudo.enable=.*@sudo.enable=${sudoEnable}@g" conf/common.properties
 
 # The following configurations may be commented, so ddd #* to ensure sed work correct
 # Change configuration in conf/worker.properties
+workerTenantAutoCreate=$(charEscape ${workerTenantAutoCreate})
+alertServer=$(charEscape ${alertServer})
 sed -i ${txt} "s@^#*worker.tenant.auto.create=.*@worker.tenant.auto.create=${workerTenantAutoCreate}@g" conf/worker.properties
 sed -i ${txt} "s@^#*alert.listen.host=.*@alert.listen.host=${alertServer}@g" conf/worker.properties
 
 # Change configuration in conf/application-api.properties
+apiServerPort=$(charEscape ${apiServerPort})
 sed -i ${txt} "s@^#*server.port=.*@server.port=${apiServerPort}@g" conf/application-api.properties
 
 # Change configuration in conf/registry.properties
+registryServers=$(charEscape ${registryServers})
+registryPluginDir=$(charEscape ${registryPluginDir})
+registryPluginName=$(charEscape ${registryPluginName})
+sed -i ${txt} "s@^#*registry.servers=.*@registry.servers=${registryServers}@g" conf/registry.properties
 sed -i ${txt} "s@^#*registry.plugin.dir=.*@registry.plugin.dir=${installPath}/${registryPluginDir}@g" conf/registry.properties
 sed -i ${txt} "s@^#*registry.plugin.name=.*@registry.plugin.name=${registryPluginName}@g" conf/registry.properties
-sed -i ${txt} "s@^#*registry.servers=.*@registry.servers=${registryServers}@g" conf/registry.properties
```

Comment:
Why does we delete this line?

Reply:
I got your point, you just move this line to the top of this group. Please recover it to the tail, cause best practices of pull requests is only change code that have to change. @tracehh

Reply:
It is redundant

Reply:
> It is redundant

I do not think so, you just move this line from L78 to L134, we could see screenshot as below.

![image](https://user-images.githubusercontent.com/15820530/142134896-070fe42f-614a-4c72-8e8b-e83eec1c0120.png)

Reply:
> It is redundant

I do not think so, you just move this line from L78 to L134, we could see screenshot as below.

![image](https://user-images.githubusercontent.com/15820530/142134896-070fe42f-614a-4c72-8e8b-e83eec1c0120.png)